packages/core/src/operator/operator-runtime/listenForChallengesCallback.ts

```diff
@@ -1,40 +1,39 @@
-import axios from "axios";
-import { checkRefereeBulkSubmissionCompatible, loadOperatorKeysFromGraph_V1, loadOperatorKeysFromRPC_V1, loadOperatorWalletsFromGraph, loadOperatorWalletsFromRPC, operatorState, processClosedChallenge, processClosedChallenges_V1, processNewChallenge, processNewChallenge_V1, PublicNodeBucketInformation } from "../index.js";
+import { checkRefereeBulkSubmissionCompatible, loadOperatorKeysFromGraph_V1, loadOperatorKeysFromRPC_V1, loadOperatorWalletsFromGraph, loadOperatorWalletsFromRPC, operatorState, processClosedChallenge, processClosedChallenges_V1, processNewChallenge, processNewChallenge_V1, PublicNodeBucketInformation, validateConfirmData } from "../index.js";
 import { Challenge, config, getSentryWalletsForOperator, getSubgraphHealthStatus, retry } from "../../index.js";
+import { ethers } from "ethers";
 
 /**
  * Update the status message for display in the operator desktop app key list
  * @param {bigint} nodeLicenseId - The nodeLicense key id.
  * @param {NodeLicenseStatus | string} newStatus - The new status to display
  */
-export async function listenForChallengesCallback(challengeNumber: bigint, challenge: Challenge, event?: any) {
-
-    if (event && challenge.rollupUsed === config.rollupAddress) {
-        compareWithCDN(challenge)
-            .then(({ publicNodeBucket, error }) => {
-                if (error) {
-                    operatorState.onAssertionMissMatchCb(publicNodeBucket, challenge, error);
-                    return;
-                }
-                operatorState.cachedLogger(`Comparison between PublicNode and Challenger was successful.`);
-            })
-            .catch(error => {
-                operatorState.cachedLogger(`Error on CND check for challenge ${Number(challenge.assertionId)}.`);
-                operatorState.cachedLogger(`${error.message}.`);
-            });
-    }
-
-    operatorState.cachedLogger(`Received new challenge with number: ${challengeNumber}. Delayed challenges will still accrue rewards.`);
-    operatorState.cachedLogger(`Processing challenge...`);
+export async function listenForChallengesCallback(challengeNumber: bigint, challenge: Challenge, event?: ethers.EventLog) {
 
     // Add a delay of 1 -300 seconds to the new challenge process so not all operators request the subgraph at the same time
     const delay = Math.floor(Math.random() * 301);
+
     await new Promise((resolve) => {
         setTimeout(resolve, delay * 1000);
     })
 
+    const graphStatus = await getSubgraphHealthStatus();
+
+    operatorState.cachedLogger(`Validating confirm data...`);
+
+    const signatureValid = await validateConfirmData(challenge, graphStatus.healthy, event);
+
+    if (!signatureValid) {
+        throw new Error(`Invalid signature for challenge ${challengeNumber}`);
+    }
```

Comment:
We cannot throw an error here, operators would not submit an assertion and loose rewards, this would incentivise operators to not validate anyhitng and submit manually to still earn rewards.

Please let me know if that makes sense to you.

Reply:
Yes, it makes sense. Error throw replaced with a log of the error. Thank you!